capstone/feed-fetching/get_vt_feed.py

```diff
@@ -0,0 +1,51 @@
+import bz2
+import io
+import itertools
+import json
+import os
+import requests
+from datetime import datetime, timedelta
+from elasticsearch import Elasticsearch
+
+es = Elasticsearch('elastic-stack-instance:9200')
+apikey = os.environ.get('VTKEY')  # is there a better way to store api key?
+
+fields_allowed = {'capabilities_tags', 'first_submission_date', 'last_submission_date', 'last_analysis_stats',
+                  'meaningful_name', 'sigma_analysis_stats', 'type_description', 'type_tag', 'reputation', 'vhash',
+                  'sha1', 'sha256', 'imphash', 'md5', 'times_submitted', 'authentihash', 'tags', 'timestamp'}
+
+
+def get_feed(timedelta_mins: int):
+    filename = (datetime.utcnow() - timedelta(minutes=timedelta_mins)).strftime('%Y%m%d%H%M')
+    response = requests.get('https://www.virustotal.com/api/v3/feeds/files/{}'.format(filename),
+                            headers={'X-Apikey': apikey})
+
+    if response.status_code != 200:
+        raise 'Feed fetching for {} failed with {}: {}'.format(filename, response.status_code, response.text)  # f-strings work only in python >= 3.6  :c
+    stream = io.BytesIO(response.content)
+
+    for line in bz2.open(stream):
+        yield json.loads(line.decode())['attributes']
+
+
+def prepare_doc(feed: dict):
+    for field in list(feed.keys()):
```

Comment:
why is a list necessary?

Reply:
Because `feed.keys()` is changing while I iterate through it and it's bad. `list` makes it static. I'll add a comment about it

Reply:
that's a bad practice. You should create a different dictionary filtering the values you want:

```python
summary_feed = {k: v for k, v in feed.items() if k in fields_allowed}
```

I encourage you to measure how much time it takes to create both dictionaries by using the timeit (https://docs.python.org/3.8/library/timeit.html) module.